src/_pytest/config/__init__.py

```diff
@@ -1268,25 +1267,42 @@ def _getini(self, name: str):
         #     a_line_list = ["tests", "acceptance"]
         #   in this case, we already have a list ready to use
         #
-        if type == "pathlist":
+        if ini_type == "pathlist":
             # TODO: This assert is probably not valid in all cases.
             assert self.inifile is not None
             dp = py.path.local(self.inifile).dirpath()
+            if not value:  # we need this because otherwise mypy raises an error
```

Comment:
If mypy complains about something that would have been a `TypeError` before, let's assert it instead of ignoring.